packages/modules-sdk/src/medusa-app.ts

```diff
@@ -0,0 +1,179 @@
+import { RemoteFetchDataCallback } from "@medusajs/orchestration"
+import {
+  LoadedModule,
+  MODULE_RESOURCE_TYPE,
+  MODULE_SCOPE,
+  ModuleConfig,
+  ModuleJoinerConfig,
+  ModuleServiceInitializeOptions,
+  RemoteJoinerQuery,
+} from "@medusajs/types"
+import {
+  ContainerRegistrationKeys,
+  ModulesSdkUtils,
+  isObject,
+} from "@medusajs/utils"
+import { MODULE_PACKAGE_NAMES, Modules } from "./definitions"
+import { MedusaModule } from "./medusa-module"
+import { RemoteLink } from "./remote-link"
+import { RemoteQuery } from "./remote-query"
+
+export type MedusaModuleConfig = (Partial<ModuleConfig> | Modules)[]
+export type SharedResources = {
+  database?: ModuleServiceInitializeOptions["database"] & {
+    pool?: {
+      min?: number
+      max?: number
+      idleTimeoutMillis?: number
+      idleInTransactionSessionTimeout?: number
+    }
+  }
```

Comment:
question: Should we expose all PG options?

Reply:
Will copy all the knex definition for these options. (they don't export the type to be used as a dev dependency)